## 모듈 import

In [ ]:
from IPython.display import Image
import numpy as np
import pandas as pd

## 데이터 다운로드

In [ ]:
# 데이터셋 다운로드를 위한 패키지 설치
!pip install teddynote -q

     |████████████████████████████████| 76.3 MB 61 kB/s 
     |████████████████████████████████| 308 kB 44.2 MB/s 
     |████████████████████████████████| 80 kB 4.4 MB/s 
     |████████████████████████████████| 209 kB 44.0 MB/s 
     |████████████████████████████████| 75 kB 3.6 MB/s 
     |████████████████████████████████| 49 kB 3.4 MB/s 
     |████████████████████████████████| 112 kB 39.2 MB/s 
     |████████████████████████████████| 149 kB 45.5 MB/s 


In [ ]:
from teddynote import dataset

In [ ]:
# 데이터셋 다운로드
dataset.download('서울시대중교통')
dataset.download('서울시주민등록인구')

======= 다운로드 시작 =======

data/seoul_transportation.xlsx


  0%|          | 0.00/26.2k [00:00<?, ?B/s]


======= 다운로드 완료 =======
======= 다운로드 시작 =======

data/seoul_population.csv


  0%|          | 0.00/3.55k [00:00<?, ?B/s]


======= 다운로드 완료 =======


## Excel

[도큐먼트](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.read_excel.html)

### Excel - 불러오기

Excel 데이터를 바로 읽어들일 수 있으며, `sheet_name`을 지정하면 해당 sheet를 가져옵니다.

[참고] `pd.read_excel()`로 엑셀 데이터 로드시 에러 발생한다면 `engine='openpyxl'`을 추가

In [ ]:
excel = pd.read_excel('data/seoul_transportation.xlsx', sheet_name='철도', engine='openpyxl')

In [ ]:
excel.head()

대중교통구분  노선명      년월   승차총승객수
0    지하철  1호선  201711  8633618
1    지하철  1호선  201712  8737235
2    지하철  1호선  201801  8145989
3    지하철  1호선  201802  7273309
4    지하철  1호선  201803  8692551

In [ ]:
excel = pd.read_excel('data/seoul_transportation.xlsx', sheet_name='버스', engine='openpyxl')
excel.head()

대중교통구분      년월     승차총승객수
0     버스  201711  163443126
1     버스  201712  162521011
2     버스  201801  153335185
3     버스  201802  134768582
4     버스  201803  166177855

`sheet_name`을 None으로 지정하면, 모든 sheet를 가지고 옵니다.

가지고 올 때는 OrderedDict로 가져오며, `keys()`로 **시트명을 조회**할 수 있습니다.

In [ ]:
excel = pd.read_excel('data/seoul_transportation.xlsx', sheet_name=None, engine='openpyxl')
excel

{'버스':    대중교통구분      년월     승차총승객수
 0      버스  201711  163443126
 1      버스  201712  162521011
 2      버스  201801  153335185
 3      버스  201802  134768582
 4      버스  201803  166177855
 5      버스  201804  160452595
 6      버스  201805  164390595
 7      버스  201806  156999747
 8      버스  201807  163736112
 9      버스  201808  160240197
 10     버스  201809  151311657
 11     버스  201810  165820934
 12     버스  201811  163017758
 13     버스  201812  158049446
 14     버스  201901  153037549
 15     버스  201902  131621925
 16     버스  201903  161694445
 17     버스  201904  161900273
 18     버스  201905  166587933, '철도':     대중교통구분    노선명      년월   승차총승객수
 0      지하철    1호선  201711  8633618
 1      지하철    1호선  201712  8737235
 2      지하철    1호선  201801  8145989
 3      지하철    1호선  201802  7273309
 4      지하철    1호선  201803  8692551
 ..     ...    ...     ...      ...
 596    지하철  우이신설선  201901  1263643
 597    지하철  우이신설선  201902  1102109
 598    지하철  우이신설선  201903  1402393
 599    지하철  우이신설선  201904  

`keys()`로 포함하고 있는 시트를 조회합니다.

In [ ]:
# 시트 조회
excel.keys()

dict_keys(['철도', '버스'])

In [ ]:
excel['철도'].head()

대중교통구분  노선명      년월   승차총승객수
0    지하철  1호선  201711  8633618
1    지하철  1호선  201712  8737235
2    지하철  1호선  201801  8145989
3    지하철  1호선  201802  7273309
4    지하철  1호선  201803  8692551

In [ ]:
excel['버스'].head()

대중교통구분      년월     승차총승객수
0     버스  201711  163443126
1     버스  201712  162521011
2     버스  201801  153335185
3     버스  201802  134768582
4     버스  201803  166177855

### Excel - 저장하기

DataFrame을 Excel로 저장할 수 있으며, Excel로 저장시 **파일명**을 지정합니다.

- `index=False` 옵션은 가급적 꼭 지정하는 옵션입니다. 지정을 안하면 **index가 별도의 컬럼으로 저장**되게 됩니다.
- `sheet_name`을 지정하여, 저장할 시트의 이름을 변경할 수 있습니다.


In [ ]:
excel = pd.read_excel('data/seoul_transportation.xlsx', sheet_name='철도', engine='openpyxl')
excel.head()

대중교통구분  노선명      년월   승차총승객수
0    지하철  1호선  201711  8633618
1    지하철  1호선  201712  8737235
2    지하철  1호선  201801  8145989
3    지하철  1호선  201802  7273309
4    지하철  1호선  201803  8692551

별도의 시트명 지정 없이 저장

In [ ]:
excel.to_excel('sample.xlsx', index=True)

시트명을 **샘플**로 지정하여 저장

In [ ]:
excel.to_excel('sample1.xlsx', index=False, sheet_name='샘플')

### Excel - 여러개의 시트에 저장

여래 개의 시트에 저장하기 위해서는 **ExcelWriter를 사용**해야 합니다.

In [ ]:
writer = pd.ExcelWriter('sample2.xlsx')
excel.to_excel(writer, index=False, sheet_name='샘플1')
excel.head(2).to_excel(writer, index=False, sheet_name='샘플2')
excel.tail(2).to_excel(writer, index=False, sheet_name='샘플3')
writer.close()

## CSV (Comma Separated Values)

- 한 줄이 한 개의 행에 해당하며, 열 사이에는 **쉼표(,)를 넣어 구분**합니다.
- Excel보다는 훨씬 가볍고 **차지하는 용량이 적기 때문에 대부분의 파일데이터는 csv 형태**로 제공됩니다.

(참고) 쉼표를 찍어 놓은 금액 데이터(100,000)를 CSV에 직접 집어넣으면 나중에 해석할 때 서로 다른 열로 취급되므로 문제가 될 수 있습니다. 해결책으로 쉼표 대신 탭 문자(\t)를 구분자로 사용하는 것이다. 이러한 경우 **Tab Separated Values(TSV)**라고 부른다.

### CSV - 불러오기

In [ ]:
df = pd.read_csv('data/seoul_population.csv')

In [ ]:
df.head()

연도  자치구        세대수  ... 등록 외국인 인구 여자수 세대당 인구 65세 이상 고령자 수
0  2020   합계  4,417,954  ...       127,824   2.19    1,568,331
1  2020  종로구     75,003  ...         5,371   1.99       28,507
2  2020   중구     63,686  ...         4,841   1.97       24,495
3  2020  용산구    113,093  ...         6,492   2.03       40,247
4  2020  성동구    135,870  ...         3,912   2.16       45,603

[5 rows x 14 columns]

때때로 한글데이터를 불러올 때 다른 인코딩을 사용해야하는 경우도 있습니다.그럴 땐 `encoding` 옵션을 지정해주면 됩니다.

In [ ]:
df = pd.read_csv('data/seoul_population.csv', encoding='utf8')

In [ ]:
df.head()

연도  자치구        세대수  ... 등록 외국인 인구 여자수 세대당 인구 65세 이상 고령자 수
0  2020   합계  4,417,954  ...       127,824   2.19    1,568,331
1  2020  종로구     75,003  ...         5,371   1.99       28,507
2  2020   중구     63,686  ...         4,841   1.97       24,495
3  2020  용산구    113,093  ...         6,492   2.03       40,247
4  2020  성동구    135,870  ...         3,912   2.16       45,603

[5 rows x 14 columns]

### CSV - 큰 파일 데이터 끊어서 불러오기

데이터의 크기가 매우 큰 경우 memory에 한 번에 로드할 수 없습니다.

`chunksize`를 지정하고 `chunksize`만큼 끊어서 불어와서 처리하게 되면 용량이 매우 큰 데이터도 처리할 수 있습니다.

예시) `chunksize=10`: 10개의 데이터를 로드합니다.

In [ ]:
df = pd.read_csv('data/seoul_population.csv', chunksize=10)

In [ ]:
for d in df:
    display(d)

연도   자치구        세대수  ... 등록 외국인 인구 여자수 세대당 인구 65세 이상 고령자 수
0  2020    합계  4,417,954  ...       127,824   2.19    1,568,331
1  2020   종로구     75,003  ...         5,371   1.99       28,507
2  2020    중구     63,686  ...         4,841   1.97       24,495
3  2020   용산구    113,093  ...         6,492   2.03       40,247
4  2020   성동구    135,870  ...         3,912   2.16       45,603
5  2020   광진구    167,427  ...         7,737   2.07       50,560
6  2020  동대문구    167,598  ...         8,624   2.05       61,796
7  2020   중랑구    185,920  ...         2,907   2.12       69,447
8  2020   성북구    195,396  ...         6,055   2.24       73,770
9  2020   강북구    145,896  ...         2,191   2.11       63,422

[10 rows x 14 columns]

연도   자치구      세대수  ... 등록 외국인 인구 여자수 세대당 인구 65세 이상 고령자 수
10  2020   도봉구  139,114  ...         1,304   2.34       62,059
11  2020   노원구  218,270  ...         2,190   2.40       85,489
12  2020   은평구  213,914  ...         2,489   2.24       85,187
13  2020  서대문구  144,838  ...         6,987   2.16       54,204
14  2020   마포구  178,482  ...         6,141   2.08       54,331
15  2020   양천구  180,633  ...         1,975   2.51       65,670
16  2020   강서구  266,982  ...         3,064   2.17       88,984
17  2020   구로구  180,531  ...        12,329   2.24       70,198
18  2020   금천구  114,715  ...         7,187   2.02       39,842
19  2020  영등포구  184,714  ...        12,565   2.05       61,688

[10 rows x 14 columns]

연도  자치구      세대수  ... 등록 외국인 인구 여자수 세대당 인구 65세 이상 고령자 수
20  2020  동작구  184,229  ...         5,786   2.12       65,190
21  2020  관악구  274,811  ...         7,871   1.80       78,875
22  2020  서초구  173,502  ...         1,976   2.45       60,169
23  2020  강남구  234,872  ...         2,445   2.30       75,096
24  2020  송파구  281,959  ...         3,185   2.37       93,483
25  2020  강동구  196,499  ...         2,200   2.34       70,019

[6 rows x 14 columns]

### CSV - 저장하기

저장하는 방법은 excel과 유사합니다. 다만, csv파일 형식에는 `sheet_name` 옵션은 없습니다.

In [ ]:
df = pd.read_csv('data/seoul_population.csv')

`to_csv()`으로 csv 파일형식으로 저장할 수 있습니다.

In [ ]:
df.to_csv('sample.csv', index=False)

읽어들인 **Excel 파일도 Csv로 저장**할 수 있습니다.

In [ ]:
excel = pd.read_excel('data/seoul_transportation.xlsx', sheet_name='버스', engine='openpyxl')
excel.head()

대중교통구분      년월     승차총승객수
0     버스  201711  163443126
1     버스  201712  162521011
2     버스  201801  153335185
3     버스  201802  134768582
4     버스  201803  166177855

In [ ]:
excel.to_csv('sample1.csv', index=False)

## 연습문제

아래 코드를 실행하여 실습에 필요한 데이터셋을 다운로드 받습니다.

In [ ]:
dataset.download('판다스입출력샘플')

======= 다운로드 시작 =======

data/file_sample.xlsx


  0%|          | 0.00/37.5k [00:00<?, ?B/s]

data/file_sample1.csv


  0%|          | 0.00/4.17k [00:00<?, ?B/s]

data/file_sample2.csv


  0%|          | 0.00/3.98k [00:00<?, ?B/s]

data/file_sample3.csv


  0%|          | 0.00/3.98k [00:00<?, ?B/s]


======= 다운로드 완료 =======


- `sample`변수에 `data/file_sample.xlsx` 파일을 읽어서 로드합니다.
- 로드한 후 sheet를 모두 출력합니다.

In [ ]:
# 코드를 입력해 주세요
sample = pd.read_excel('data/file_sample.xlsx',sheet_name=None)
sample.keys()

dict_keys(['2020년 01월', '2020년 02월', '2020년 03월', '2020년 04월', '2020년 05월', '2020년 06월', '2020년 07월', '2020년 08월', '2020년 09월', '2020년 10월', '2020년 11월', '2020년 12월'])

<p><strong>[출력 결과]</strong></p><pre>dict_keys(['2020년 01월', '2020년 02월', '2020년 03월', '2020년 04월', '2020년 05월', '2020년 06월', '2020년 07월', '2020년 08월', '2020년 09월', '2020년 10월', '2020년 11월', '2020년 12월'])</pre>

2020년 10월 시트를 `sample_202010`에 로드 합니다.

In [ ]:
# 코드를 입력해 주세요
sample_202010 = sample['2020년 10월']
sample_202010

지역     GS칼텍스_셀프     GS칼텍스_일반  ...  자가상표_일반    현대오일뱅크_셀프    현대오일뱅크_일반
0    서울 강남구  1443.440000  1627.111111  ...      NaN  1385.230000  1525.705000
1    서울 강동구  1413.275000  1430.353333  ...      NaN  1403.063333  1424.340000
2    서울 강북구  1293.675000  1313.190000  ...      NaN  1322.300000          NaN
3    서울 강서구  1361.773333  1385.840000  ...      NaN  1299.240000  1362.530000
4    서울 관악구  1415.450000  1429.970000  ...      NaN  1392.550000  1407.050000
5    서울 광진구          NaN  1360.394000  ...      NaN  1358.733333  1337.940000
6    서울 구로구  1285.940000  1511.173333  ...      NaN  1314.235000  1391.170000
7    서울 금천구  1313.940000          NaN  ...      NaN  1313.035000  1379.390000
8    서울 노원구  1340.890000  1421.850000  ...      NaN          NaN  1431.435000
9    서울 도봉구  1331.920000  1327.320000  ...      NaN  1345.106667  1309.936667
10  서울 동대문구  1305.485000  1304.980000  ...      NaN  1329.100000  1386.727500
11   서울 동작구  1413.840000  1345.275000  ...      NaN  1393.290000  1388.840000
12   서울 마포구          NaN  1371.190000  ...      NaN  1453.000000  1420.580000
13  서울 서대문구  1357.690000  1358.230000  ...      NaN  1352.840000  1357.805000
14   서울 서초구  1391.770000  1408.220000  ...      NaN  1417.290000  1468.968000
15   서울 성동구  1361.450000  1650.000000  ...  1299.58  1350.725000  1572.695000
16   서울 성북구  1299.240000  1333.710000  ...      NaN  1306.835000  1353.160000
17   서울 송파구  1364.285000  1381.450000  ...      NaN  1367.140000  1378.280000
18   서울 양천구  1312.826667  1323.050000  ...      NaN  1347.365000  1401.385000
19  서울 영등포구  1350.437500  1580.468333  ...      NaN  1349.728000  1494.000000
20   서울 용산구          NaN  1769.362500  ...      NaN          NaN  1800.507500
21   서울 은평구  1332.586000  1341.520000  ...  1418.00  1314.870000  1399.000000
22   서울 종로구          NaN  1989.000000  ...      NaN          NaN  1665.775000
23    서울 중구          NaN  2067.053333  ...      NaN          NaN  1618.000000
24   서울 중랑구  1315.525000  1305.710000  ...      NaN  1338.420000  1291.190000

[25 rows x 13 columns]

<p><strong>[출력 결과]</strong></p><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>지역</th>
      <th>GS칼텍스_셀프</th>
      <th>GS칼텍스_일반</th>
      <th>S-OIL_셀프</th>
      <th>S-OIL_일반</th>
      <th>SK에너지_셀프</th>
      <th>SK에너지_일반</th>
      <th>알뜰(ex)_셀프</th>
      <th>알뜰주유소_셀프</th>
      <th>알뜰주유소_일반</th>
      <th>자가상표_일반</th>
      <th>현대오일뱅크_셀프</th>
      <th>현대오일뱅크_일반</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>서울 강남구</td>
      <td>1443.440000</td>
      <td>1627.111111</td>
      <td>1420.760000</td>
      <td>1593.250</td>
      <td>1416.866250</td>
      <td>1734.166667</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>1385.230000</td>
      <td>1525.705000</td>
    </tr>
    <tr>
      <th>1</th>
      <td>서울 강동구</td>
      <td>1413.275000</td>
      <td>1430.353333</td>
      <td>1418.970000</td>
      <td>1498.000</td>
      <td>1445.000000</td>
      <td>1653.695000</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>1403.063333</td>
      <td>1424.340000</td>
    </tr>
    <tr>
      <th>2</th>
      <td>서울 강북구</td>
      <td>1293.675000</td>
      <td>1313.190000</td>
      <td>1275.100000</td>
      <td>1325.710</td>
      <td>1329.690000</td>
      <td>1295.840000</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>1322.300000</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>3</th>
      <td>서울 강서구</td>
      <td>1361.773333</td>
      <td>1385.840000</td>
      <td>1376.913333</td>
      <td>NaN</td>
      <td>1325.516667</td>
      <td>1433.664000</td>
      <td>NaN</td>
      <td>1273.11</td>
      <td>1264.06</td>
      <td>NaN</td>
      <td>1299.240000</td>
      <td>1362.530000</td>
    </tr>
    <tr>
      <th>4</th>
      <td>서울 관악구</td>
      <td>1415.450000</td>
      <td>1429.970000</td>
      <td>NaN</td>
      <td>1397.015</td>
      <td>1411.840000</td>
      <td>1436.280000</td>
      <td>NaN</td>
      <td>1313.00</td>
      <td>1329.45</td>
      <td>NaN</td>
      <td>1392.550000</td>
      <td>1407.050000</td>
    </tr>
    <tr>
      <th>5</th>
      <td>서울 광진구</td>
      <td>NaN</td>
      <td>1360.394000</td>
      <td>1337.160000</td>
      <td>1344.970</td>
      <td>NaN</td>
      <td>1373.623333</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>1309.19</td>
      <td>NaN</td>
      <td>1358.733333</td>
      <td>1337.940000</td>
    </tr>
    <tr>
      <th>6</th>
      <td>서울 구로구</td>
      <td>1285.940000</td>
      <td>1511.173333</td>
      <td>1286.420000</td>
      <td>1393.832</td>
      <td>NaN</td>
      <td>1336.065000</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>1321.77</td>
      <td>NaN</td>
      <td>1314.235000</td>
      <td>1391.170000</td>
    </tr>
    <tr>
      <th>7</th>
      <td>서울 금천구</td>
      <td>1313.940000</td>
      <td>NaN</td>
      <td>1305.565000</td>
      <td>1287.580</td>
      <td>NaN</td>
      <td>1351.600000</td>
      <td>NaN</td>
      <td>1282.46</td>
      <td>1291.67</td>
      <td>NaN</td>
      <td>1313.035000</td>
      <td>1379.390000</td>
    </tr>
    <tr>
      <th>8</th>
      <td>서울 노원구</td>
      <td>1340.890000</td>
      <td>1421.850000</td>
      <td>1370.966667</td>
      <td>NaN</td>
      <td>1373.275000</td>
      <td>1398.000000</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>1431.435000</td>
    </tr>
    <tr>
      <th>9</th>
      <td>서울 도봉구</td>
      <td>1331.920000</td>
      <td>1327.320000</td>
      <td>1329.896000</td>
      <td>1557.000</td>
      <td>1291.470000</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>1345.106667</td>
      <td>1309.936667</td>
    </tr>
    <tr>
      <th>10</th>
      <td>서울 동대문구</td>
      <td>1305.485000</td>
      <td>1304.980000</td>
      <td>1296.390000</td>
      <td>1372.500</td>
      <td>1367.550000</td>
      <td>1383.838000</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>1329.100000</td>
      <td>1386.727500</td>
    </tr>
    <tr>
      <th>11</th>
      <td>서울 동작구</td>
      <td>1413.840000</td>
      <td>1345.275000</td>
      <td>NaN</td>
      <td>1358.030</td>
      <td>NaN</td>
      <td>1484.350000</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>1393.290000</td>
      <td>1388.840000</td>
    </tr>
    <tr>
      <th>12</th>
      <td>서울 마포구</td>
      <td>NaN</td>
      <td>1371.190000</td>
      <td>1408.970000</td>
      <td>1556.175</td>
      <td>NaN</td>
      <td>1493.846000</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>1453.000000</td>
      <td>1420.580000</td>
    </tr>
    <tr>
      <th>13</th>
      <td>서울 서대문구</td>
      <td>1357.690000</td>
      <td>1358.230000</td>
      <td>1351.905000</td>
      <td>1320.810</td>
      <td>1360.226667</td>
      <td>1352.713333</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>1352.840000</td>
      <td>1357.805000</td>
    </tr>
    <tr>
      <th>14</th>
      <td>서울 서초구</td>
      <td>1391.770000</td>
      <td>1408.220000</td>
      <td>1388.463333</td>
      <td>1629.940</td>
      <td>1402.270000</td>
      <td>1590.833333</td>
      <td>1272.39</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>1417.290000</td>
      <td>1468.968000</td>
    </tr>
    <tr>
      <th>15</th>
      <td>서울 성동구</td>
      <td>1361.450000</td>
      <td>1650.000000</td>
      <td>1312.230000</td>
      <td>1342.190</td>
      <td>1418.770000</td>
      <td>1471.978333</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>1299.58</td>
      <td>1350.725000</td>
      <td>1572.695000</td>
    </tr>
    <tr>
      <th>16</th>
      <td>서울 성북구</td>
      <td>1299.240000</td>
      <td>1333.710000</td>
      <td>1301.643333</td>
      <td>1354.725</td>
      <td>1309.065000</td>
      <td>1407.230000</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>1263.84</td>
      <td>NaN</td>
      <td>1306.835000</td>
      <td>1353.160000</td>
    </tr>
    <tr>
      <th>17</th>
      <td>서울 송파구</td>
      <td>1364.285000</td>
      <td>1381.450000</td>
      <td>1356.130000</td>
      <td>1412.840</td>
      <td>1359.053333</td>
      <td>1431.812500</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>1367.140000</td>
      <td>1378.280000</td>
    </tr>
    <tr>
      <th>18</th>
      <td>서울 양천구</td>
      <td>1312.826667</td>
      <td>1323.050000</td>
      <td>1269.060000</td>
      <td>NaN</td>
      <td>1333.678000</td>
      <td>1447.435000</td>
      <td>NaN</td>
      <td>1292.87</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>1347.365000</td>
      <td>1401.385000</td>
    </tr>
    <tr>
      <th>19</th>
      <td>서울 영등포구</td>
      <td>1350.437500</td>
      <td>1580.468333</td>
      <td>1349.940000</td>
      <td>1345.000</td>
      <td>1338.742500</td>
      <td>1600.968333</td>
      <td>NaN</td>
      <td>1276.42</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>1349.728000</td>
      <td>1494.000000</td>
    </tr>
    <tr>
      <th>20</th>
      <td>서울 용산구</td>
      <td>NaN</td>
      <td>1769.362500</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>1885.408333</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>1800.507500</td>
    </tr>
    <tr>
      <th>21</th>
      <td>서울 은평구</td>
      <td>1332.586000</td>
      <td>1341.520000</td>
      <td>1338.050000</td>
      <td>NaN</td>
      <td>1334.837500</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>1418.00</td>
      <td>1314.870000</td>
      <td>1399.000000</td>
    </tr>
    <tr>
      <th>22</th>
      <td>서울 종로구</td>
      <td>NaN</td>
      <td>1989.000000</td>
      <td>NaN</td>
      <td>1459.000</td>
      <td>1425.000000</td>
      <td>1943.000000</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>1665.775000</td>
    </tr>
    <tr>
      <th>23</th>
      <td>서울 중구</td>
      <td>NaN</td>
      <td>2067.053333</td>
      <td>NaN</td>
      <td>1758.000</td>
      <td>NaN</td>
      <td>1945.522000</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>1401.87</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>1618.000000</td>
    </tr>
    <tr>
      <th>24</th>
      <td>서울 중랑구</td>
      <td>1315.525000</td>
      <td>1305.710000</td>
      <td>1303.493333</td>
      <td>1359.885</td>
      <td>1347.587500</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>1338.420000</td>
      <td>1291.190000</td>
    </tr>
  </tbody>
</table>
</div>

- `sample_202010`을 **2020-10-oil-price.csv**로 내보냅니다.
- index는 지정하지 않습니다.

In [ ]:
# 코드를 입력해 주세요
sample_202010.to_csv('2020-10-oil-price.csv',index=False)

- **2020-10-oil-price.csv**로 저장한 파일을 `sample_csv`로 로드합니다.
- 단 5개의 행으로 끊어서 로드하여 반복문을 활용하여 출력합니다.
- 출력은 `print()` 대신 `display()`를 활용하여 출력합니다.

In [ ]:
# 코드를 입력해 주세요
sample_csv = pd.read_csv('2020-10-oil-price.csv',chunksize=5)

for each in sample_csv:
  display(each)

지역     GS칼텍스_셀프     GS칼텍스_일반  ...  자가상표_일반    현대오일뱅크_셀프  현대오일뱅크_일반
0  서울 강남구  1443.440000  1627.111111  ...      NaN  1385.230000   1525.705
1  서울 강동구  1413.275000  1430.353333  ...      NaN  1403.063333   1424.340
2  서울 강북구  1293.675000  1313.190000  ...      NaN  1322.300000        NaN
3  서울 강서구  1361.773333  1385.840000  ...      NaN  1299.240000   1362.530
4  서울 관악구  1415.450000  1429.970000  ...      NaN  1392.550000   1407.050

[5 rows x 13 columns]

지역  GS칼텍스_셀프     GS칼텍스_일반  ...  자가상표_일반    현대오일뱅크_셀프    현대오일뱅크_일반
5  서울 광진구       NaN  1360.394000  ...      NaN  1358.733333  1337.940000
6  서울 구로구   1285.94  1511.173333  ...      NaN  1314.235000  1391.170000
7  서울 금천구   1313.94          NaN  ...      NaN  1313.035000  1379.390000
8  서울 노원구   1340.89  1421.850000  ...      NaN          NaN  1431.435000
9  서울 도봉구   1331.92  1327.320000  ...      NaN  1345.106667  1309.936667

[5 rows x 13 columns]

지역  GS칼텍스_셀프  GS칼텍스_일반  ...  자가상표_일반  현대오일뱅크_셀프  현대오일뱅크_일반
10  서울 동대문구  1305.485  1304.980  ...      NaN    1329.10  1386.7275
11   서울 동작구  1413.840  1345.275  ...      NaN    1393.29  1388.8400
12   서울 마포구       NaN  1371.190  ...      NaN    1453.00  1420.5800
13  서울 서대문구  1357.690  1358.230  ...      NaN    1352.84  1357.8050
14   서울 서초구  1391.770  1408.220  ...      NaN    1417.29  1468.9680

[5 rows x 13 columns]

지역     GS칼텍스_셀프     GS칼텍스_일반  ...  자가상표_일반  현대오일뱅크_셀프  현대오일뱅크_일반
15   서울 성동구  1361.450000  1650.000000  ...  1299.58   1350.725   1572.695
16   서울 성북구  1299.240000  1333.710000  ...      NaN   1306.835   1353.160
17   서울 송파구  1364.285000  1381.450000  ...      NaN   1367.140   1378.280
18   서울 양천구  1312.826667  1323.050000  ...      NaN   1347.365   1401.385
19  서울 영등포구  1350.437500  1580.468333  ...      NaN   1349.728   1494.000

[5 rows x 13 columns]

지역  GS칼텍스_셀프     GS칼텍스_일반  ...  자가상표_일반  현대오일뱅크_셀프  현대오일뱅크_일반
20  서울 용산구       NaN  1769.362500  ...      NaN        NaN  1800.5075
21  서울 은평구  1332.586  1341.520000  ...   1418.0    1314.87  1399.0000
22  서울 종로구       NaN  1989.000000  ...      NaN        NaN  1665.7750
23   서울 중구       NaN  2067.053333  ...      NaN        NaN  1618.0000
24  서울 중랑구  1315.525  1305.710000  ...      NaN    1338.42  1291.1900

[5 rows x 13 columns]

<p><strong>[출력 결과]</strong></p><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>지역</th>
      <th>GS칼텍스_셀프</th>
      <th>GS칼텍스_일반</th>
      <th>S-OIL_셀프</th>
      <th>S-OIL_일반</th>
      <th>SK에너지_셀프</th>
      <th>SK에너지_일반</th>
      <th>알뜰(ex)_셀프</th>
      <th>알뜰주유소_셀프</th>
      <th>알뜰주유소_일반</th>
      <th>자가상표_일반</th>
      <th>현대오일뱅크_셀프</th>
      <th>현대오일뱅크_일반</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>서울 강남구</td>
      <td>1443.440000</td>
      <td>1627.111111</td>
      <td>1420.760000</td>
      <td>1593.250</td>
      <td>1416.866250</td>
      <td>1734.166667</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>1385.230000</td>
      <td>1525.705</td>
    </tr>
    <tr>
      <th>1</th>
      <td>서울 강동구</td>
      <td>1413.275000</td>
      <td>1430.353333</td>
      <td>1418.970000</td>
      <td>1498.000</td>
      <td>1445.000000</td>
      <td>1653.695000</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>1403.063333</td>
      <td>1424.340</td>
    </tr>
    <tr>
      <th>2</th>
      <td>서울 강북구</td>
      <td>1293.675000</td>
      <td>1313.190000</td>
      <td>1275.100000</td>
      <td>1325.710</td>
      <td>1329.690000</td>
      <td>1295.840000</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>1322.300000</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>3</th>
      <td>서울 강서구</td>
      <td>1361.773333</td>
      <td>1385.840000</td>
      <td>1376.913333</td>
      <td>NaN</td>
      <td>1325.516667</td>
      <td>1433.664000</td>
      <td>NaN</td>
      <td>1273.11</td>
      <td>1264.06</td>
      <td>NaN</td>
      <td>1299.240000</td>
      <td>1362.530</td>
    </tr>
    <tr>
      <th>4</th>
      <td>서울 관악구</td>
      <td>1415.450000</td>
      <td>1429.970000</td>
      <td>NaN</td>
      <td>1397.015</td>
      <td>1411.840000</td>
      <td>1436.280000</td>
      <td>NaN</td>
      <td>1313.00</td>
      <td>1329.45</td>
      <td>NaN</td>
      <td>1392.550000</td>
      <td>1407.050</td>
    </tr>
  </tbody>
</table>
</div>

<p><strong>[출력 결과]</strong></p><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>지역</th>
      <th>GS칼텍스_셀프</th>
      <th>GS칼텍스_일반</th>
      <th>S-OIL_셀프</th>
      <th>S-OIL_일반</th>
      <th>SK에너지_셀프</th>
      <th>SK에너지_일반</th>
      <th>알뜰(ex)_셀프</th>
      <th>알뜰주유소_셀프</th>
      <th>알뜰주유소_일반</th>
      <th>자가상표_일반</th>
      <th>현대오일뱅크_셀프</th>
      <th>현대오일뱅크_일반</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>5</th>
      <td>서울 광진구</td>
      <td>NaN</td>
      <td>1360.394000</td>
      <td>1337.160000</td>
      <td>1344.970</td>
      <td>NaN</td>
      <td>1373.623333</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>1309.19</td>
      <td>NaN</td>
      <td>1358.733333</td>
      <td>1337.940000</td>
    </tr>
    <tr>
      <th>6</th>
      <td>서울 구로구</td>
      <td>1285.94</td>
      <td>1511.173333</td>
      <td>1286.420000</td>
      <td>1393.832</td>
      <td>NaN</td>
      <td>1336.065000</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>1321.77</td>
      <td>NaN</td>
      <td>1314.235000</td>
      <td>1391.170000</td>
    </tr>
    <tr>
      <th>7</th>
      <td>서울 금천구</td>
      <td>1313.94</td>
      <td>NaN</td>
      <td>1305.565000</td>
      <td>1287.580</td>
      <td>NaN</td>
      <td>1351.600000</td>
      <td>NaN</td>
      <td>1282.46</td>
      <td>1291.67</td>
      <td>NaN</td>
      <td>1313.035000</td>
      <td>1379.390000</td>
    </tr>
    <tr>
      <th>8</th>
      <td>서울 노원구</td>
      <td>1340.89</td>
      <td>1421.850000</td>
      <td>1370.966667</td>
      <td>NaN</td>
      <td>1373.275</td>
      <td>1398.000000</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>1431.435000</td>
    </tr>
    <tr>
      <th>9</th>
      <td>서울 도봉구</td>
      <td>1331.92</td>
      <td>1327.320000</td>
      <td>1329.896000</td>
      <td>1557.000</td>
      <td>1291.470</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>1345.106667</td>
      <td>1309.936667</td>
    </tr>
  </tbody>
</table>
</div>

<p><strong>[출력 결과]</strong></p><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>지역</th>
      <th>GS칼텍스_셀프</th>
      <th>GS칼텍스_일반</th>
      <th>S-OIL_셀프</th>
      <th>S-OIL_일반</th>
      <th>SK에너지_셀프</th>
      <th>SK에너지_일반</th>
      <th>알뜰(ex)_셀프</th>
      <th>알뜰주유소_셀프</th>
      <th>알뜰주유소_일반</th>
      <th>자가상표_일반</th>
      <th>현대오일뱅크_셀프</th>
      <th>현대오일뱅크_일반</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>10</th>
      <td>서울 동대문구</td>
      <td>1305.485</td>
      <td>1304.980</td>
      <td>1296.390000</td>
      <td>1372.500</td>
      <td>1367.550000</td>
      <td>1383.838000</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>1329.10</td>
      <td>1386.7275</td>
    </tr>
    <tr>
      <th>11</th>
      <td>서울 동작구</td>
      <td>1413.840</td>
      <td>1345.275</td>
      <td>NaN</td>
      <td>1358.030</td>
      <td>NaN</td>
      <td>1484.350000</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>1393.29</td>
      <td>1388.8400</td>
    </tr>
    <tr>
      <th>12</th>
      <td>서울 마포구</td>
      <td>NaN</td>
      <td>1371.190</td>
      <td>1408.970000</td>
      <td>1556.175</td>
      <td>NaN</td>
      <td>1493.846000</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>1453.00</td>
      <td>1420.5800</td>
    </tr>
    <tr>
      <th>13</th>
      <td>서울 서대문구</td>
      <td>1357.690</td>
      <td>1358.230</td>
      <td>1351.905000</td>
      <td>1320.810</td>
      <td>1360.226667</td>
      <td>1352.713333</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>1352.84</td>
      <td>1357.8050</td>
    </tr>
    <tr>
      <th>14</th>
      <td>서울 서초구</td>
      <td>1391.770</td>
      <td>1408.220</td>
      <td>1388.463333</td>
      <td>1629.940</td>
      <td>1402.270000</td>
      <td>1590.833333</td>
      <td>1272.39</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>1417.29</td>
      <td>1468.9680</td>
    </tr>
  </tbody>
</table>
</div>

<p><strong>[출력 결과]</strong></p><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>지역</th>
      <th>GS칼텍스_셀프</th>
      <th>GS칼텍스_일반</th>
      <th>S-OIL_셀프</th>
      <th>S-OIL_일반</th>
      <th>SK에너지_셀프</th>
      <th>SK에너지_일반</th>
      <th>알뜰(ex)_셀프</th>
      <th>알뜰주유소_셀프</th>
      <th>알뜰주유소_일반</th>
      <th>자가상표_일반</th>
      <th>현대오일뱅크_셀프</th>
      <th>현대오일뱅크_일반</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>15</th>
      <td>서울 성동구</td>
      <td>1361.450000</td>
      <td>1650.000000</td>
      <td>1312.230000</td>
      <td>1342.190</td>
      <td>1418.770000</td>
      <td>1471.978333</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>1299.58</td>
      <td>1350.725</td>
      <td>1572.695</td>
    </tr>
    <tr>
      <th>16</th>
      <td>서울 성북구</td>
      <td>1299.240000</td>
      <td>1333.710000</td>
      <td>1301.643333</td>
      <td>1354.725</td>
      <td>1309.065000</td>
      <td>1407.230000</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>1263.84</td>
      <td>NaN</td>
      <td>1306.835</td>
      <td>1353.160</td>
    </tr>
    <tr>
      <th>17</th>
      <td>서울 송파구</td>
      <td>1364.285000</td>
      <td>1381.450000</td>
      <td>1356.130000</td>
      <td>1412.840</td>
      <td>1359.053333</td>
      <td>1431.812500</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>1367.140</td>
      <td>1378.280</td>
    </tr>
    <tr>
      <th>18</th>
      <td>서울 양천구</td>
      <td>1312.826667</td>
      <td>1323.050000</td>
      <td>1269.060000</td>
      <td>NaN</td>
      <td>1333.678000</td>
      <td>1447.435000</td>
      <td>NaN</td>
      <td>1292.87</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>1347.365</td>
      <td>1401.385</td>
    </tr>
    <tr>
      <th>19</th>
      <td>서울 영등포구</td>
      <td>1350.437500</td>
      <td>1580.468333</td>
      <td>1349.940000</td>
      <td>1345.000</td>
      <td>1338.742500</td>
      <td>1600.968333</td>
      <td>NaN</td>
      <td>1276.42</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>1349.728</td>
      <td>1494.000</td>
    </tr>
  </tbody>
</table>
</div>

<p><strong>[출력 결과]</strong></p><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>지역</th>
      <th>GS칼텍스_셀프</th>
      <th>GS칼텍스_일반</th>
      <th>S-OIL_셀프</th>
      <th>S-OIL_일반</th>
      <th>SK에너지_셀프</th>
      <th>SK에너지_일반</th>
      <th>알뜰(ex)_셀프</th>
      <th>알뜰주유소_셀프</th>
      <th>알뜰주유소_일반</th>
      <th>자가상표_일반</th>
      <th>현대오일뱅크_셀프</th>
      <th>현대오일뱅크_일반</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>20</th>
      <td>서울 용산구</td>
      <td>NaN</td>
      <td>1769.362500</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>1885.408333</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>1800.5075</td>
    </tr>
    <tr>
      <th>21</th>
      <td>서울 은평구</td>
      <td>1332.586</td>
      <td>1341.520000</td>
      <td>1338.050000</td>
      <td>NaN</td>
      <td>1334.8375</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>1418.0</td>
      <td>1314.87</td>
      <td>1399.0000</td>
    </tr>
    <tr>
      <th>22</th>
      <td>서울 종로구</td>
      <td>NaN</td>
      <td>1989.000000</td>
      <td>NaN</td>
      <td>1459.000</td>
      <td>1425.0000</td>
      <td>1943.000000</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>1665.7750</td>
    </tr>
    <tr>
      <th>23</th>
      <td>서울 중구</td>
      <td>NaN</td>
      <td>2067.053333</td>
      <td>NaN</td>
      <td>1758.000</td>
      <td>NaN</td>
      <td>1945.522000</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>1401.87</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>1618.0000</td>
    </tr>
    <tr>
      <th>24</th>
      <td>서울 중랑구</td>
      <td>1315.525</td>
      <td>1305.710000</td>
      <td>1303.493333</td>
      <td>1359.885</td>
      <td>1347.5875</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>1338.42</td>
      <td>1291.1900</td>
    </tr>
  </tbody>
</table>
</div>